internal/adapters/operation/operation_storage_redis_test.go

```diff
@@ -876,3 +876,146 @@ func TestCleanOperationsHistory(t *testing.T) {
+						Event:     "some-event",
+					},
+				},
+			},
+			{
+				ID:             "some-op-id-4",
+				SchedulerName:  schedulerName,
+				Status:         operation.StatusFinished,
+				DefinitionName: "test-definition",
+				CreatedAt:      now.Add(-29 * time.Hour),
+				Input:          []byte("hello test"),
+				ExecutionHistory: []operation.OperationEvent{
+					{
+						CreatedAt: time.Date(1999, time.November, 19, 6, 12, 15, 0, time.UTC),
+						Event:     "some-event",
+					},
+				},
+			},
+		}
+
+		t.Run("if operations history is not empty it deletes all expired operations", func(t *testing.T) {
+			client := test.GetRedisConnection(t, redisAddress)
+			clock := clockmock.NewFakeClock(time.Now())
+			operationsTTlMap := map[Definition]time.Duration{}
+			storage := NewRedisOperationStorage(client, clock, operationsTTlMap)
+
+			for _, op := range operations {
+				executionHistoryJson, err := json.Marshal(op.ExecutionHistory)
+				require.NoError(t, err)
+
+				err = client.ZAdd(context.Background(), storage.buildSchedulerHistoryOperationsKey(op.SchedulerName), &redis.Z{
+					Member: op.ID,
+					Score:  float64(op.CreatedAt.Unix()),
+				}).Err()
+				require.NoError(t, err)
+
+				for _, op := range operations[0:1] {
+					err = client.HSet(context.Background(), storage.buildSchedulerOperationKey(op.SchedulerName, op.ID), map[string]interface{}{
+						idRedisKey:                 op.ID,
+						schedulerNameRedisKey:      op.SchedulerName,
+						statusRedisKey:             strconv.Itoa(int(op.Status)),
+						definitionNameRedisKey:     op.DefinitionName,
+						createdAtRedisKey:          op.CreatedAt.Format(time.RFC3339Nano),
+						definitionContentsRedisKey: op.Input,
+						executionHistoryRedisKey:   executionHistoryJson,
+					}).Err()
+					require.NoError(t, err)
+				}
+
+				err = storage.CleanExpiredOperations(context.Background(), schedulerName)
+				assert.NoError(t, err)
+
+				operationIds, err := client.ZRangeByScore(context.Background(), fmt.Sprintf("operations:%s:lists:history", schedulerName), &redis.ZRangeBy{
+					Min: "-inf",
+					Max: "+inf",
+				}).Result()
+				require.NoError(t, err)
+				assert.NotEmpty(t, operationIds)
+				assert.Contains(t, operationIds, operations[0].ID, operations[1].ID)
+				assert.NotContains(t, operationIds, operations[2].ID, operations[3].ID)
+			}
+		})
+
+		t.Run("if operations history is empty it returns with no error", func(t *testing.T) {
+			client := test.GetRedisConnection(t, redisAddress)
+			clock := clockmock.NewFakeClock(time.Now())
+			operationsTTlMap := map[Definition]time.Duration{}
+			storage := NewRedisOperationStorage(client, clock, operationsTTlMap)
+
+			operationsReturned, err := storage.ListSchedulerFinishedOperations(context.Background(), schedulerName)
+			assert.NoError(t, err)
+			assert.Empty(t, operationsReturned)
+
+			err = storage.CleanExpiredOperations(context.Background(), schedulerName)
+
+			assert.NoError(t, err)
```

Comment:
Hey @mflilian I am missing here an assertion that ensures that there is no operation on the sorted set. I think that doing something like the last test test before:
```golang
                        operationIds, err := client.ZRangeByScore(context.Background(), fmt.Sprintf("operations:%s:lists:history", schedulerName), &redis.ZRangeBy{
					Min: "-inf",
					Max: "+inf",
				}).Result()
				require.NoError(t, err)
				assert.NotEmpty(t, operationIds)
				assert.Contains(t, operationIds, operations[0].ID, operations[1].ID)
				assert.NotContains(t, operationIds, operations[2].ID, operations[3].ID)
			}
```